src/components/[guild]/RoleCard/components/Reward.tsx

```diff
@@ -197,53 +199,62 @@ const RewardIcon = ({
   withMotionImg = true,
   transition,
 }: RewardIconProps) => {
-  const circleBgColor = useColorModeValue("gray.700", "gray.600")
-
+  const [doIconFallback, setDoIconFallback] = useState(false)
   const props = {
     src:
-      guildPlatform?.platformGuildData?.imageUrl ??
-      rewards[PlatformType[guildPlatform?.platformId]]?.imageUrl,
-    alt: guildPlatform?.platformGuildName,
+      guildPlatform.platformGuildData?.imageUrl ??
+      rewards[PlatformType[guildPlatform.platformId]].imageUrl,
+    alt: guildPlatform.platformGuildName,
     boxSize: 6,
     rounded: "full",
-  }
-
-  const circleProps = {
-    bgColor: circleBgColor,
-    boxSize: 6,
+    onError: () => {
+      setDoIconFallback(true)
+    },
   }
 
   const motionElementProps = {
     layoutId: `${rolePlatformId}_reward_img`,
     transition: { type: "spring", duration: 0.5, ...transition },
   }
+  const circleBgColor = useColorModeValue("gray.700", "gray.600")
+  const circleProps = {
+    bgColor: circleBgColor,
+    boxSize: 6,
+  }
 
-  if (!props.src) {
-    if (withMotionImg)
+  if (doIconFallback) {
```

Comment:
I think we could keep the `!props.src` condition here too, what do you think? (In that case users wouldn't need to fetch the image, then fallback to the icon if it errors out)